operators/pkg/controller/apmserverelasticsearchassociation/apmserverelasticsearchassociation_controller.go

```diff
@@ -172,6 +173,12 @@ func (r *ReconcileApmServerElasticsearchAssociation) Reconcile(request reconcile
 		if err := r.Status().Update(&apmServer); err != nil {
 			return defaultRequeue, err
 		}
+		r.recorder.AnnotatedEventf(&apmServer,
+			annotation.ForAssociationStatusChange(origStatus.Association, newStatus),
+			corev1.EventTypeNormal,
+			events.EventAssociationStatusChange,
+			"Association status changed from [%s] to [%s]", origStatus.Association, newStatus)
```

Comment:
Is it possible that origStatus.Association and newStatus are the same? if check is done on ApmServerStatus and not AssociationStatus. While not part of this change, the naming here could be improved to distinguish between those two.

Reply:
That's a good question. I suppose there is a subtle race condition here because the association controller could overwrite the status calculated by the APM controller and vice versa. I already had some concerns about the use of `reflect.DeepEqual` in the reconciliation loops as well so I'll raise a separate issue to look into it.